File: src/main/java/io/github/chaosdave34/kitpvp/abilities/impl/magician/ShuffleAbility.java
package io.github.chaosdave34.kitpvp.abilities.impl.magician;

import io.github.chaosdave34.kitpvp.abilities.Ability;

import net.kyori.adventure.text.Component;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class ShuffleAbility extends Ability {
    public ShuffleAbility() {
        super("shuffle", "Shuffle", Type.SNEAK_RIGHT_CLICK, 5);
    }

    @Override
    public @NotNull List<Component> getDescription() {
        return createSimpleDescription(
                "Shuffles the inventory of all players",
                "in a 5 block radius."
        );
    }

    @Override
    public boolean onAbility(Player p) {
        for (Entity entity : p.getNearbyEntities(5, 5, 5)) {
            if (entity instanceof Player target) {
                PlayerInventory inventory = target.getInventory();
                List<ItemStack> shuffledContent = new ArrayList<>(List.of(inventory.getContents()));
                Collections.shuffle(shuffledContent);
                inventory.setContents(shuffledContent.toArray(new ItemStack[]{}));
            }
        }
        return true;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/abilities/impl/magician/ShuffleAbility.kt
package io.github.chaosdave34.kitpvp.abilities.impl.magician

import io.github.chaosdave34.kitpvp.abilities.Ability
import net.kyori.adventure.text.Component
import org.bukkit.entity.HumanEntity
import org.bukkit.entity.Player

class ShuffleAbility : Ability("shuffle", "Shuffle", Type.RIGHT_CLICK, 10) {
    override fun getDescription(): List<Component> = createSimpleDescriptionAsList("Shuffles the inventory the enemy you are looking at in a 8 block radius.")

    override fun onAbility(player: Player): Boolean {
        val target = player.getTargetEntity(8)
        if (target is HumanEntity && target.checkTargetIfPlayer()) {
            val inventory = target.inventory
            val content = mutableListOf(*inventory.contents)
            val shuffledContent = content.subList(0, 36)
            val rest = content.subList(36, content.size)
            shuffledContent.shuffle()
            shuffledContent.addAll(rest)
            inventory.contents = shuffledContent.toTypedArray()
            return true
        }
        return false
    }
}